docs/user/security/authentication/index.asciidoc

```diff
@@ -119,6 +119,19 @@ The order of `saml` and `basic` is important. Users who open {kib} will go throu
 
 Basic authentication is supported _only_ if `basic` authentication provider is explicitly declared in `xpack.security.authc.providers` setting in addition to `saml`.
 
+[float]
+===== SAML and long URLs
+
+At the beginning of the SAML handshake {kib} stores initial URL in the session cookie to be able to redirect user back to this URL after successful SAML authentication.
```

Comment:
```suggestion
At the beginning of the SAML handshake {kib}, stores the initial URL in the session cookie, so it can redirect the user back to that URL after successful SAML authentication.
```